jax/_src/numpy/lax_numpy.py

```diff
@@ -2271,8 +2271,16 @@ def identity(n: DimSize, dtype: Optional[DTypeLike] = None) -> Array:
 def arange(start: DimSize, stop: Optional[DimSize] = None,
            step: Optional[DimSize] = None, dtype: Optional[DTypeLike] = None) -> Array:
   dtypes.check_user_dtype_supported(dtype, "arange")
-  require = partial(core.concrete_or_error, None)
-  msg = "It arose in jax.numpy.arange argument `{}`.".format
+  if not jax.config.jax_dynamic_shapes:
+    util.check_arraylike_or_none("arange", start, stop, step)
```

Comment:
Maybe use the `check_arraylike` variant for `start` as it gives better error message for `jnp.arange(None)`.

Reply:
Done, thanks